src/Generators/Microsoft.Gen.MetricsReports/MetricDefinitionEmitter.cs

```diff
@@ -13,70 +13,94 @@ namespace Microsoft.Gen.MetricsReports;
 
 internal sealed class MetricDefinitionEmitter : EmitterBase
 {
+    private const int DimensionsIndentLevel = 2;
+
     internal MetricDefinitionEmitter()
         : base(false)
     {
     }
 
-    public string GenerateReport(IReadOnlyList<ReportedMetricClass> metricClasses, CancellationToken cancellationToken)
+    /// <summary>
+    /// Generates JSON object containing the <see cref="ReportedMetricClass"/> for metrics report.
+    /// </summary>
+    /// <param name="metricClasses">The reported metric classes.</param>
+    /// <param name="cancellationToken">The cancellation token.</param>
+    /// <param name="indentationLevel">The number of indentations in case its nested in other reports like <see cref="MetadataReportsGenerator"/>.Defaulted to zero.</param>
+    /// <returns>string report as json or String.Empty.</returns>
+    public string GenerateReport(IReadOnlyList<ReportedMetricClass> metricClasses, CancellationToken cancellationToken = default, int indentationLevel = 0)
     {
         if (metricClasses == null || metricClasses.Count == 0)
         {
             return string.Empty;
         }
 
+        if (indentationLevel > 0)
+        {
+            OutLn();
+        }
+
+        Indent(indentationLevel);
         OutLn("[");
 
         for (int i = 0; i < metricClasses.Count; i++)
         {
             cancellationToken.ThrowIfCancellationRequested();
             var metricClass = metricClasses[i];
+
             GenMetricClassDefinition(metricClass, cancellationToken);
 
             if (i < metricClasses.Count - 1)
             {
-                Out(",");
+                OutLn(",");
             }
-
-            OutLn();
         }
 
-        Out("]");
+        OutLn("]");
+        Unindent(indentationLevel);
+
         return Capture();
     }
 
     private void GenMetricClassDefinition(ReportedMetricClass metricClass, CancellationToken cancellationToken)
     {
         cancellationToken.ThrowIfCancellationRequested();
-        OutLn(" {");
-
-        OutLn($"  \"{metricClass.RootNamespace}\":");
+        Indent();
+        OutLn("{");
```

Comment:
```suggestion
        OutOpenBrace();
```

ditto for any other Indent / Outln("{"} calls